purchase_vendor_promotion/models/stock_warehouse_orderpoint.py

```diff
@@ -12,6 +12,25 @@ class StockWarehouseOrderpoint(models.Model):
     )
     promotion_date_start = fields.Date(compute="_compute_promotion", store=True)
     promotion_date_end = fields.Date(compute="_compute_promotion", store=True)
+    supplier_id = fields.Many2one(
+        compute="_compute_supplier_id", readonly=False, store=True
+    )
+
+    @api.depends("route_id", "route_id.force_vendor_with_best_promotion")
+    def _compute_supplier_id(self):
+        for rec in self:
+            if rec.route_id and rec.route_id.force_vendor_with_best_promotion:
+                allowed_company_ids = rec.allowed_location_ids.mapped("company_id")
+                suppliers = rec.product_id._prepare_sellers(False).filtered(
+                    lambda x: x.company_id in allowed_company_ids
+                )
+                promotion_suppliers = suppliers.filtered(
+                    lambda x: x._is_promotion_active_or_upcoming()
+                )
```

Comment:
Is there a reason to use 2 separated `filtered()` instead of a single one combining the 2 conditions?

Reply:
The idea is to make `suppliers` acts as the fallback when there is no `promotion_suppliers`. 
So, the second filtered was used to filtered based on records of suppliers, instead of querying all again.

Any recommendation is welcome